### PR TITLE
hack/validate/vendor: print diff for modified files

### PR DESCRIPTION
**Pull request by kolyshkin**
In case some files were modified (rather than merely added
or removed), we're curious to see the diff for those.

Idea-by: @thaJeztah 

While at it,
* fix some shellcheck warnings;
* optimize/simplify `validate_vendor_used`.

### CODE REVIEW

File: hack/validate/vendor [hunk 1]
```diff
@@ -36,9 +40,9 @@ validate_vendor_diff(){
 validate_vendor_used() {
 	pkgs=$(mawk '/^[a-zA-Z0-9]/ { print $1 }' < vendor.conf)
 	for f in $pkgs; do
-	if ls -d vendor/$f  > /dev/null 2>&1; then
-		found=$(find vendor/$f -iregex '.*LICENSE.*' -or -iregex '.*COPYRIGHT.*' -or -iregex '.*COPYING.*' | wc -l)
-		if [ $found -eq 0 ]; then
+	if ls -d "vendor/$f"  > /dev/null 2>&1; then
+		found=$(find "vendor/$f" -iregex '.*LICENSE.*' -or -iregex '.*COPYRIGHT.*' -or -iregex '.*COPYING.*' | wc -l)
```
Comment (thaJeztah) [inline hack/validate/vendor]: Not for this PR, wondering if those regexes could be combined (haven't tried), i.e.

```
'.*(LICENSE|COPYRIGHT|COPYING).*'
```

(guess "NOTICE" should also be in there)

**Review by thaJeztah — APPROVED** [review]
LGTM 👍

File: hack/validate/vendor (hunk 1 above)
Comment (kolyshkin) [reply hack/validate/vendor]: I resisted the urge to rewrite this, yes. 

Alsop, `.*` at the beginning and the end is not needed.

Comment (kolyshkin) [reply hack/validate/vendor]: added a commit optimizing it

**Comment by kolyshkin** [comment]
@thaJeztah first two commits remain the same (so you won't have to re-review), the only change is I added the third one with `validate_vendor_used` as you suggested.

**Comment by kolyshkin** [comment]
> Should we add NOTICE? IIUC, the notice is the actual copyright/licensing information, and having a license file is merely for convenience

1. Currently, we don't have any packages that fail validation (i.e. they all either container COPYING or LICENSE in some way).

1. Let's have a look,

```console
$ grep -vEc '^$|^#' vendor.conf 
118
$ find vendor -name NOTICE\* | wc -l
13
```
...out of 118 packages that moby/moby currently vendors, only 13 packages contain NOTICE, and about half of those NOTICEs only contain a copyright statement but not any licensing information. I guess the aim of this check we are actually checking for licensing terms (to make sure it's compatible with moby's own licensing terms) not copyright.

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by andrewhsu — APPROVED** [review]
SGTM

would be nice to have an example run to show the effects of this change working; regardless, i already see the value of this PR in diagnosing certain situations.

**Review by tonistiigi — APPROVED** [review]

File: hack/validate/vendor [hunk 2]
```diff
@@ -13,6 +13,7 @@ validate_vendor_diff(){
 		vndr
 		# check if any files have changed
 		diffs="$(git status --porcelain -- vendor 2>/dev/null)"
```
Comment (SamWhited) [inline hack/validate/vendor]: Should this check `vendor.conf` as well?

Comment (thaJeztah) [reply hack/validate/vendor]: vendor.conf is only updated manually, so that shouldn't be an issue

**Pull request merged**